design/step-target-safe-mode.md

```diff
@@ -0,0 +1,87 @@
+# Step target command "safe mode"
+
+With the introduction of [step targets](step-target.md), the code running inside a given step can be mostly isolated from the agent host.
+Agent logging commands remain a vector for untrusted user code to make permanent changes to its environment.
+For instance, `task.setVariable` can alter system variables for the duration of the job.
+`artifact.upload` can exfiltrate bits off the machine or poison otherwise-trusted artifacts.
+
+This spec proposes an extension to step targets allowing them to be placed in "command safe mode".
+In command safe mode, only a small, approved subset of logging commands are processed.
+These are necessary for communicating the status or outcome of a step.
+All other commands are excluded.
+
+## What gets isolated?
+
+By default, all commands are denied.
+In order to be allowed, a command should meet all of these criteria:
+- Command is necessary to update the agent about the current step's progress or state
+- There is no safer way to perform the action
+- After reasonable threat modeling, we can't see a way for untrusted code to exploit the command
+
+## Syntax
+
+The `target` property of a step will be expanded to allow either a simple string or a mapping.
+If it's a string, the string is the target of the step (either `host` or a container resource name) and safe mode is off (as if `commands: any` were specified).
+If it's a mapping, the following keys are supported:
+
+```yaml
+- script: ...
+  target:
+    container: string (container name or the word `host`)
+    commands: enum (one of "safe" or "any", with "any" the default)
```

Comment:
`safe` might infer incorrectly that the alternative is `unsafe`

How about `restricted` instead?